### PR TITLE
Fix Convective Baseboard with Hard Sizes

### LINKED ISSUE #10769: EnergyPlus 24.2 produces zero heating energy for hardsized baseboard
Issue overview
--------------
For the attached model with a hardsized boiler, E+ 24.2 produces zero heating energy use while E+ 24.1 produced non-zero heating energy use. E+ 24.2 also throws a new warning:

```
   ** Warning ** SizePlantLoop: Calculated Plant Sizing Design Volume Flow Rate=[0.00] is too small. Set to 0.0
   **   ~~~   ** ..occurs for PlantLoop=BOILER HYDRONIC HEAT LOOP
```

and the eio file shows these diffs:

image

@mjwitte thinks it is probably related to https://github.com/NREL/EnergyPlus/pull/10721.

Full set of input/output files:
[ep241_242.zip](https://github.com/user-attachments/files/17150942/ep241_242.zip)

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to EnergyPlus Defect Complexity (Github Project)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by mjwitte**
Pull request overview
---------------------
 - Fixes #10769
 - #10721 fixed a crash with baseboard sizing when the baseboard is hard-sized and no zone sizing calculations were done. The prior fix was to skip the baseboard sizing function. But this skipped some eio size reporting and it skipped the call to `PlantUtilities::RegisterPlantCompDesignFlow` which is needed if the baseboard is autosized or hard-sized.
 - The fix here always calls `BaseboardParams::SizeBaseboard` and moves some reference assignments to avoid the crash. Also, the `resetSizingFlagBasedOnInput` function is changed to `checkForZoneSizing` which calls `CheckZoneSizing` is anything is autosized.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [x] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [x] If feature, test running new feature, try creative ways to break it
 - [x] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/BaseboardRadiator.cc [hunk 1]
```diff
@@ -638,6 +636,8 @@ namespace BaseboardRadiator {
                             state, cCMO_BBRadiator_Water, this->EquipID, "User-Specified Maximum Water Flow Rate [m3/s]", this->WaterVolFlowRateMax);
                     }
                 } else {
+                    auto &zoneEqSizing = state.dataSize->ZoneEqSizing(state.dataSize->CurZoneEqNum);
+                    auto const &finalZoneSizing = state.dataSize->FinalZoneSizing(state.dataSize->CurZoneEqNum);
```
Comment (rraustad) [inline src/EnergyPlus/BaseboardRadiator.cc:640]: I wonder why I keep missing these allocation issues. Even here this doesn't look right. At this else, the flow could be autosized and a zone sizing run might not be requested. I think there needs to be a CheckZoneSizing call here to fatal out before the sizing arrays are accessed.

    if (!FlowAutoSize && !state.dataSize->ZoneSizingRunDone) {
    } else {
        std::string_view const CompType = cCMO_BBRadiator_Water;
        std::string_view const CompName = this->EquipID;
        CheckZoneSizing(state, CompType, CompName);
        // now OK to access sizing arrays
        auto &zoneEqSizing = state.dataSize->ZoneEqSizing(state.dataSize->CurZoneEqNum);
        auto const &finalZoneSizing = state.dataSize->FinalZoneSizing(state.dataSize->CurZoneEqNum);

Comment (mjwitte) [reply src/EnergyPlus/BaseboardRadiator.cc:640]: Several `CheckZoneSizing` calls were removed in #10721 and consolidated in the precheck function which is now called `checkForZoneSizing`.

Comment (rraustad) [reply src/EnergyPlus/BaseboardRadiator.cc:640]: Right. And these references still need to be located where sizing is requested so this is good.

**Comment by rraustad** [comment]
I tried to break this without success. The closest I got was reporting baseboard sizing when a Sizing:Plant object exists and otherwise not reporting that data. Since the UA and water flow rate are hard sized this is probably not a big deal. I don't think there is anything to do here so just documenting the fact that several variations of inputs were tested and passed.

    ZoneHVAC:Baseboard:Convective:Water,
      Zone1Baseboard,          !- Name
      FanAndCoilAvailSched,    !- Availability Schedule Name
      Zone1BBHWInletNode,      !- Inlet Node Name
      Zone1BBHWOutletNode,     !- Outlet Node Name
      CapacityPerFloorArea,   !- Heating Design Capacity Method
      autosize,                !- Heating Design Capacity {W}
      50.,                        !- Heating Design Capacity Per Floor Area {W/m2}
      1.2,                        !- Fraction of Autosized Heating Design Capacity
      500.,                    !- U-Factor Times Area Value {W/K}
      0.0012,                  !- Maximum Water Flow Rate {m3/s}
      0.001;                   !- Convergence Tolerance

[TryToBreak-in.idf.txt](https://github.com/user-attachments/files/17171938/TryToBreak-in.idf.txt)

**Review by rraustad — APPROVED** [review]
These changes look sound. Unit test shows expected severe/fatal.

**Comment by Myoldmopar** [comment]
Documentation issue is obviously unrelated.  I'm doing a quick build locally to verify things.

**Comment by Myoldmopar** [comment]
This one is also happy locally, let's get this in as well.  Thanks for this one @mjwitte

**Pull request merged**